salt/modules/win_lgpo.py

```diff
@@ -4156,13 +4170,13 @@ def _write_regpol_data(data_to_write,
             ret = __salt__['file.makedirs'](policy_file_path)
         with salt.utils.files.fopen(policy_file_path, 'wb') as pol_file:
             if not data_to_write.startswith(reg_pol_header.encode('utf-16-le')):
-                pol_file.write(reg_pol_header.encode('utf-16-le'))
-            pol_file.write(data_to_write)
+                pol_file.write(salt.utils.stringutils.to_bytes(reg_pol_header.encode('utf-16-le')))
+            pol_file.write(salt.utils.stringutils.to_bytes(data_to_write))
         try:
             gpt_ini_data = ''
             if os.path.exists(gpt_ini_path):
                 with salt.utils.files.fopen(gpt_ini_path, 'rb') as gpt_file:
-                    gpt_ini_data = salt.utils.stringutils.to_str(gpt_file.read())
+                    gpt_ini_data = salt.utils.stringutils.to_unicode(gpt_file.read())
```

Comment:
We've had some strange encodings in some files... But I think it was because there was actually some malformed data in the file... One of the files is actually `utf-16-le`. Not sure which one. I think it's the registry.pol file. @lomeroe would know.

Reply:
I agree that this should be okay for gpt.ini

registry.pol should be read/written raw, as it has different pieces in different encodings within the file and attempting to convert some to unicode will result in the unicode surrogate errors (see #44944)

Reply:
:+1:  Thanks everyone.  I'll update this line.